include/deal.II/base/mpi_consensus_algorithm.h

```diff
@@ -0,0 +1,205 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_mpi_consensus_algorithm_h
+#define dealii_mpi_consensus_algorithm_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/mpi.h>
+#include <deal.II/base/mpi.templates.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+
+namespace Utilities
+{
+  namespace MPI
+  {
+    /**
+     * Anonymous version of the Utilities::MPI::ConsensusAlgorithmProcess.
```

Comment:
```suggestion
     * An anonymous version of the Utilities::MPI::ConsensusAlgorithmProcess.
```
Also, for me, it's not obvious what "anonymous" means in this context (without you explaining in the pull request discussion). I would just say that this class implements `Utilities::MPI::ConsensusAlgorithmProcess` using user-provided function wrappers. Of course, lambdas are a good example.